packages/mobile/src/app/App.tsx

```diff
@@ -52,32 +57,42 @@ BigNumber.config({
 // }
 
 export class App extends React.Component {
+  state: State = {
+    reactInitTime: undefined,
+    reactLoadTime: undefined,
+  }
+
+  appStartListener: EmitterSubscription = new NativeEventEmitter().addListener(
+    'AppStartedLoading',
+    (reactInitTime: string) => {
+      this.setState({
+        reactInitTime: +reactInitTime,
+        reactLoadTime: Date.now(),
+      })
+    }
+  )
+
   async componentDidMount() {
     await ValoraAnalytics.init()
-    const appLoadedMillis: number = Date.now()
     const { width, height } = Dimensions.get('window')
 
-    if (Platform.OS === 'android') {
-      const appStartListener = DeviceEventEmitter.addListener(
-        'AppStartedLoading',
-        (appStartedMillisStr: string) => {
-          const appStartedMillis: number = +appStartedMillisStr
-          const loadingDuration = appLoadedMillis - appStartedMillis
-          ValoraAnalytics.startSession(AppEvents.app_launched, {
-            loadingDuration,
-            deviceHeight: height,
-            deviceWidth: width,
-          })
-          appStartListener.remove()
-        }
-      )
-    } else {
-      ValoraAnalytics.startSession(AppEvents.app_launched, {
-        deviceHeight: height,
-        deviceWidth: width,
-      })
+    let reactLoadDuration
+    let appLoadDuration
+
+    if (this.state.reactInitTime && this.state.reactLoadTime) {
+      const appLoadedTime = Date.now()
+      reactLoadDuration = (this.state.reactLoadTime - this.state.reactInitTime) / 1000
+      appLoadDuration = (appLoadedTime - this.state.reactInitTime) / 1000
     }
 
+    ValoraAnalytics.startSession(AppEvents.app_launched, {
```

Comment:
Thoughts on a separate app_loaded event with the loadDuration payload, to make it easier to query where those values are non-null? This would also allow for const assignment of reactLoadDuration and appLoadDuration